packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -157,31 +166,46 @@ export class ForkChoice implements IForkChoice {
 
   getHead(): IBlockSummary {
     // balances is not changed but votes are changed
-    if (!this.synced) {
-      const deltas = computeDeltas(this.protoArray.indices, this.votes, this.justifiedBalances, this.justifiedBalances);
-      this.protoArray.applyScoreChanges(
-        deltas,
-        this.fcStore.justifiedCheckpoint.epoch,
-        this.fcStore.finalizedCheckpoint.epoch
-      );
-      this.synced = true;
-    }
-    const headRoot = this.protoArray.findHead(toHexString(this.fcStore.justifiedCheckpoint.root));
-    const headIndex = this.protoArray.indices.get(headRoot);
-    if (headIndex === undefined) {
-      throw new ForkChoiceError({
-        code: ForkChoiceErrorCode.MISSING_PROTO_ARRAY_BLOCK,
-        root: fromHexString(headRoot),
-      });
-    }
-    const headNode = this.protoArray.nodes[headIndex];
-    if (headNode === undefined) {
-      throw new ForkChoiceError({
-        code: ForkChoiceErrorCode.MISSING_PROTO_ARRAY_BLOCK,
-        root: fromHexString(headRoot),
-      });
+    let timer;
+    try {
+      if (!this.synced) {
+        timer = this.metrics?.forkChoiceFindHead.startTimer();
+        const deltas = computeDeltas(
+          this.protoArray.indices,
+          this.votes,
+          this.justifiedBalances,
+          this.justifiedBalances
+        );
+        this.protoArray.applyScoreChanges(
+          deltas,
+          this.fcStore.justifiedCheckpoint.epoch,
+          this.fcStore.finalizedCheckpoint.epoch
+        );
+        this.synced = true;
+      }
+      const headRoot = this.protoArray.findHead(toHexString(this.fcStore.justifiedCheckpoint.root));
+      const headIndex = this.protoArray.indices.get(headRoot);
+      if (headIndex === undefined) {
+        throw new ForkChoiceError({
+          code: ForkChoiceErrorCode.MISSING_PROTO_ARRAY_BLOCK,
+          root: fromHexString(headRoot),
+        });
+      }
+      const headNode = this.protoArray.nodes[headIndex];
+      if (headNode === undefined) {
+        throw new ForkChoiceError({
+          code: ForkChoiceErrorCode.MISSING_PROTO_ARRAY_BLOCK,
+          root: fromHexString(headRoot),
+        });
+      }
+
+      const blockSummary = toBlockSummary(headNode);
+      if (timer) timer();
+      return blockSummary;
+    } catch (e) {
+      if (timer) timer();
+      throw e;
```

Comment:
Oh nice idea to stop the time on error! you could do that in a finally block and won't have to re-throw